hubbard/plot/plot.py

```diff
@@ -276,6 +277,9 @@ def real_space_grid(v, grid_unit, density):
 
         slice_grid = grid.grid[:, :, 0].T.real
 
+        if mode in ['density']:
+            slice_grid = slice_grid**2
```

Comment:
Hmm, I see what you were trying to do. But I think the arguments are misleading and needs more thinking. When I read `density` I think of charge density of the full system, not a single eigenstate.

Reply:
Maybe I can use 'mode=DOS' instead, what do you think?

Reply:
Hmm... No I don't think so. Hmm.. In this case I would probably use `LDOS`.
And `density` and `charge` would be the same. ?

Reply:
OK, so use `if mode in ['charge', 'density']` for what we call charge plots (i.e. spinpolarization, charge, chargedif... etc)
and change what we call here mode='density' for mode='LDOS', no? I will do these changes and upload it now.

Reply:
To me that would make more sense, but perhaps @tfrederiksen could chip in...

Reply:
Yes, precisely. In the case of `DOS` plots we are squaring a grid that is not computed from the density matrix. The only cases where we use this grid is for charge plots (spin polarization, charge, charge diff, etc.).
For the `DOS` plots we now are using the grid computed from `sisl.wavefunction` and then squaring it to compute the DOS.

Reply:
What about a more generic internal option `square_gridvalues=False/True` to describe explicitly how the passed physical quantity is handled on the real-space grid?

Reply:
That is also fine for me, although in this case we would have to use two flags instead of only one (the flag `mode` (or whatever we want to call it) would still be needed to specify whether we need the grid to be computed from the WF or from the DM).

Reply:
Can't the function not simply check internally whether the input is of type WF or DM?

Reply:
Using the flag `mode` was my approach to do it :)
The thing is that in many cases the input is simply a vector (it can also be a `sisl.physics.electron.EigenstateElectron` instance e.g. for WF plots) which then is expanded in the real-space grid (where the way in which the real-space grid is computed is determined by the flag `mode` in this case).